src/helpers/dataTransfer.ts

```diff
@@ -0,0 +1,120 @@
+import { XYPosition } from 'react-flow-renderer';
+import log from 'electron-log';
+import { extname } from 'path';
+import { NodeProto } from './contexts/GlobalNodeState';
+import { SchemaMap } from './SchemaMap';
+import { FileOpenResult, ipcRenderer } from './safeIpc';
+import { SaveFile } from './SaveFile';
+
+export interface ChainnerDragData {
+    schemaId: string;
+    offsetX?: number;
+    offsetY?: number;
+}
+
+export const enum TransferTypes {
+    ChainerSchema = 'application/chainner/schema',
+}
+
+export interface DataTransferProcessorOptions {
+    createNode: (proto: NodeProto) => void;
+    getNodePosition: (offsetX?: number, offsetY?: number) => XYPosition;
+    schemata: SchemaMap;
+}
+
+export const openSaveFile = async (path: string): Promise<FileOpenResult> => {
+    try {
+        const saveData = await SaveFile.read(path);
+        return { kind: 'Success', path, saveData };
+    } catch (error) {
+        log.error(error);
+        return { kind: 'Error', path, error: String(error) };
+    }
+};
+
+/**
+ * Returns `false` if the data couldn't not be processed by this processor.
```

Comment:
```suggestion
 * Returns `false` if the data could not be processed by this processor.
```

Reply:
I hate that I sometimes just leave out words while writing. Thank you for spotting this.